File: server/src/main/java/bit/project/server/entity/Saleitem.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
public class Saleitem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;


    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    private Sale sale;


    @ManyToOne
    private Item item;

    private Integer qty;

    private Integer unitprice;

    public Saleitem(Integer id) {
        this.id = id;
    }
}

File: client/src/app/entities/saleitem.ts
import {Sale} from "./sale";
import {Item} from "./item";

export class Saleitem {
  id: number;
  sale: Sale;
  item: Item;
  qty: number;
  unitprice: number;


  constructor(id: number=null
  ) {
    this.id = id;
  }
}
